common/event_emitter_test.go

```diff
@@ -125,3 +127,187 @@ func TestEventEmitterAllEvents(t *testing.T) {
 		})
 	})
 }
+
+//nolint:gocognit
+func TestBaseEventEmitter(t *testing.T) {
+	t.Parallel()
+
+	t.Run("order of emitted events kept", func(t *testing.T) {
+		// Test description
+		//
+		// 1. Emit many events from the emitWorker.
+		// 2. Handler receives the emitted events.
+		//
+		// Success criteria: Ensure that the ordering of events is
+		//                   received in the order they're emitted.
+
+		t.Parallel()
+
+		eventName := "AtomicIntEvent"
+		maxInt := 100
+
+		ctx, cancel := context.WithCancel(context.Background())
+		emitter := NewBaseEventEmitter(ctx)
+		ch := make(chan Event)
+		emitter.on(ctx, []string{eventName}, ch)
+
+		var expectedI int
+		handler := func() {
+			defer cancel()
+
+			for expectedI != maxInt {
+				e := <-ch
+
+				i, ok := e.data.(int)
+				if !ok {
+					assert.FailNow(t, "unexpected type read from channel", e.data)
+				}
+
+				assert.Equal(t, eventName, e.typ)
+				assert.Equal(t, expectedI, i)
+
+				expectedI++
+			}
+
+			close(ch)
+		}
+		go handler()
+
+		emitWorker := func() {
+			for i := 0; i < maxInt; i++ {
+				i := i
```

Comment:
There is no need for this.

Reply:
Yeah, you're correct. No need for var shadowing here.